tempesta_fw/t/unit/test_tfw_str.c

```diff
@@ -314,101 +343,58 @@ TEST(tfw_str_eq_cstr, handles_empty_strs)
 
 TEST(tfw_str_eq_cstr, supports_casei)
 {
-	TFW_STR(s, "FooBarBaz 123");
+	DEFINE_TFW_STR(s, "FooBarBaz 123");
 	const char *cstr1 = "FooBarBaz 123";
 	const char *cstr2 = "fooBarBaz 123";
 	const char *cstr3 = "FooBarBaZ 123";
 	size_t len1 = strlen(cstr1);
 	size_t len2 = strlen(cstr2);
 	size_t len3 = strlen(cstr3);
-
-	EXPECT_TRUE(tfw_str_eq_cstr(s, cstr1, len1, TFW_STR_EQ_CASEI));
-	EXPECT_TRUE(tfw_str_eq_cstr(s, cstr2, len2, TFW_STR_EQ_CASEI));
-	EXPECT_TRUE(tfw_str_eq_cstr(s, cstr3, len3, TFW_STR_EQ_CASEI));
-	EXPECT_FALSE(tfw_str_eq_cstr(s, cstr2, len2, TFW_STR_EQ_DEFAULT));
-	EXPECT_FALSE(tfw_str_eq_cstr(s, cstr3, len3, TFW_STR_EQ_DEFAULT));
+	TFW_DBG("test_cassei:s:%s\n", s.data);
+	EXPECT_TRUE(tfw_str_eq_cstr(&s, cstr1, len1, TFW_STR_EQ_CASEI));
+	EXPECT_TRUE(tfw_str_eq_cstr(&s, cstr2, len2, TFW_STR_EQ_CASEI));
+	EXPECT_TRUE(tfw_str_eq_cstr(&s, cstr3, len3, TFW_STR_EQ_CASEI));
+	EXPECT_FALSE(tfw_str_eq_cstr(&s, cstr2, len2, TFW_STR_EQ_DEFAULT));
+	EXPECT_FALSE(tfw_str_eq_cstr(&s, cstr3, len3, TFW_STR_EQ_DEFAULT));
```

Comment:
This is a strange piece of code. All these strings are constant. Why not define them as `TfwStr{}` strings with the help of `DEFINE_TFW_STR()` and use the `data` and `len` members of each corresponding string? That way all data is defined at compile time. There's not need to compute the length of each string with `strlen()` at run time.